crypto/bls12381_utils.c

```diff
@@ -189,36 +189,51 @@ void bn_randZr_star(bn_t x) {
     byte seed[seed_len];
     rand_bytes(seed, seed_len);
     bn_map_to_Zr_star(x, seed, seed_len);
+    rand_bytes(seed, seed_len); // overwrite seed
 }
 
 // generates a random number less than the order r
 void bn_randZr(bn_t x) {
-    bn_t r;
-    bn_new(r); 
-    g2_get_ord(r);
     // reduce the modular reduction bias
     bn_new_size(x, BITS_TO_DIGITS(Fr_BITS + SEC_BITS));
     bn_rand(x, RLC_POS, Fr_BITS + SEC_BITS);
-    bn_mod(x, x, r);
-    bn_free(r);
+    bn_mod(x, x, &core_get()->ep_r);
 }
 
 // reads a scalar from an array and maps it to Zr
+// the resulting scalar is in the range 0 <= a < r
+// len must be less than BITS_TO_BYTES(RLC_BN_BITS).
```

Comment:
> the resulting scalar is in the range 0 <= a < r

is `a` the scalar? Hard to differentiate between the symbol and the english word "a". 


```suggestion
// reads a scalar from an array and maps it to Zr;
// the resulting scalar `a` is in the range 0 <= a < r;
// len must be less than BITS_TO_BYTES(RLC_BN_BITS).
```